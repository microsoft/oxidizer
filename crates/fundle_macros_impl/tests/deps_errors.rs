// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use syn::{parse_quote, ItemStruct, ItemUnion};

mod util;

#[test]
fn empty() {
    let item: ItemStruct = parse_quote! {
        #[deps]
        struct Foo {}
    };

    insta::assert_snapshot!(expand_fundle_deps!(item));
}

#[test]
fn tuple() {
    let item: ItemStruct = parse_quote! {
        #[deps]
        struct Foo(u8);
    };

    insta::assert_snapshot!(expand_fundle_deps!(item));
}

#[test]
fn unit() {
    let item: ItemStruct = parse_quote! {
        #[deps]
        struct Foo;
    };

    insta::assert_snapshot!(expand_fundle_deps!(item));
}


#[test]
fn union() {
    let item: ItemUnion = parse_quote! {
        #[deps]
        union Foo {}
    };

    insta::assert_snapshot!(expand_fundle_deps!(item));
}