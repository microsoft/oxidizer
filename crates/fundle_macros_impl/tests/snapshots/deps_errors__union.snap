---
source: crates/fundle_macros_impl/tests/deps_errors.rs
assertion_line: 25
expression: expand_fundle_deps!(item)
---
::core::compile_error! {
    "fundle::args can only be applied to structs"
}
